File: app/src/main/java/com/quran/labs/androidquran/ui/util/PageController.java
package com.quran.labs.androidquran.ui.util;

import android.view.MotionEvent;

import com.quran.data.model.SuraAyah;
import com.quran.labs.androidquran.ui.helpers.AyahSelectedListener;

public interface PageController {
  boolean handleTouchEvent(MotionEvent event, AyahSelectedListener.EventType eventType, int page);
  void handleRetryClicked();
  void onScrollChanged(float y);
  void handleLongPress(SuraAyah suraAyah);
  void endAyahMode();
}

File: app/src/main/java/com/quran/labs/androidquran/ui/util/PageController.kt
package com.quran.labs.androidquran.ui.util

import android.view.MotionEvent
import com.quran.data.model.SuraAyah
import com.quran.labs.androidquran.ui.helpers.AyahSelectedListener

interface PageController {
  fun handleTouchEvent(
    event: MotionEvent,
    eventType: AyahSelectedListener.EventType,
    page: Int
  ): Boolean

  fun handleRetryClicked()
  fun onScrollChanged(y: Float)
  fun handleLongPress(suraAyah: SuraAyah)
  fun endAyahMode()
}
